compiler/parser/parser.peg

```diff
@@ -778,8 +778,11 @@ Primary
   / "(" __ expr:Expr __ ")" { RETURN(expr) }
 
 Record
-  = "{" __ fields:Fields __ "}" {
-      RETURN(MAP("kind":"RecordExpr", "fields":fields ))
+  = "{" __ expr:Expr _ "with" _ fields:Fields __ "}" {
+      RETURN(MAP("kind":"RecordExpr", "fields":fields, "with":expr))
+    }
+  / "{" __ fields:Fields __ "}" {
+      RETURN(MAP("kind":"RecordExpr", "fields":fields, "with":NULL))
```

Comment:
```suggestion
  = "{" __ with_:(e:Expr _ "with" _ { RETURN(e) })? fields:Fields __ "}" {
      RETURN(MAP("kind":"RecordExpr", "fields":fields, "with":with_))
```

Reply:
I find this pattern a little harder to follow and maintain especially when debugging backtracking problems so I'd like to leave it in the explanded form.